src/main/java/sefariaproject/TextController.java

```diff
@@ -0,0 +1,67 @@
+package sefariaproject;
+
+import io.reactivex.rxjava3.schedulers.Schedulers;
+import sefariaproject.text.Text;
+
+import javax.swing.*;
+
+public class TextController {
+
+    private String perek;
+    private JTextArea textArea;
+    private Text text;
+    private SefariaService service;
+
+    public TextController(SefariaService service, JTextArea textArea) {
+        this.textArea = textArea;
+        this.service = service;
+    }
+
+    public void updateText(String textName) {
+        service.getSefariaText(textName)
+                .subscribeOn(Schedulers.io())
+                .observeOn(Schedulers.newThread())
```

Comment:
Do not use `newThread()` use what I am now using here
https://github.com/andrewoid/schwimmer-weather-2023/pull/4